GoogleForCreators/web-stories-wp
# Pre-Publish Checklist: Highlight and focus inputs

## Summary
- show highlights and focus inputs from the prepublish checklist



## Relevant Technical Choices
- created a new context selector hook which components can use to determine how to show highlights and when to focus inner refs
- there are no styles in figma associated with this. Had some back and forth with design to decide on the look. This may be used in the future with the context/quick menus.
- had to forward the ref on `input` and `textArea`
- Focused anything that made sense to focus. This gives us some automatic screen reading when issues are selected, the reader will also let the user know when the button/input to correct the issue is focused as well.

**Here's what it looks like now**
These images are available in the design document. That's also where design uploaded the highlight effect visuals.

![accessibility](https://user-images.githubusercontent.com/41136059/106504793-a9133100-6484-11eb-876c-10131d8b90c5.gif)
![assistive text](https://user-images.githubusercontent.com/41136059/106504798-aa445e00-6484-11eb-8fb8-52df03fe1f5a.gif)
![video captions](https://user-images.githubusercontent.com/41136059/106504802-ac0e2180-6484-11eb-8798-d480a3515b56.gif)
![storytitle](https://user-images.githubusercontent.com/41136059/106504811-add7e500-6484-11eb-9ab6-615b3d9b180a.gif)
![story cover](https://user-images.githubusercontent.com/41136059/106504817-afa1a880-6484-11eb-9578-b9536c89a377.gif)



## To-do
- minimum karma tests:
- [x] opens the inspector tab
- [x] opens the text tab
- [x] focuses input



## User-facing changes
- more prepublish checklist items will be underlined. 
- clicking on newly underlined checklist items will open the correct inspector tab, highlight the panels, and focus the buttons/inputs.


## Testing Instructions
- create a new story
- open prepublish checklist tab
- click on "missing story cover"
- document tab should open and highlight the cover image input
- more expected behavior is [documented here](https://docs.google.com/document/d/1soSgYqJoIRwiyWZxs2FBVTuw2-EQf4KJQQG3HBnUhM4/edit)



---



Fixes #5631 
Fixes #6081
